src/Microsoft.QualityGuidelines.Analyzers/Core/SealMethodsThatSatisfyPrivateInterfaces.cs

```diff
@@ -39,13 +41,15 @@ public override void Initialize(AnalysisContext analysisContext)
 
             analysisContext.RegisterSymbolAction(CheckTypes, SymbolKind.NamedType);
         }
-
+        
         private static void CheckTypes(SymbolAnalysisContext context)
         {
-            var type = (ITypeSymbol)context.Symbol;
+            var type = (INamedTypeSymbol)context.Symbol;
 
-            // only classes can have overridable members
-            if (type.TypeKind == TypeKind.Class)
+            // Only classes can have overridable members, and furthermore, only consider classes that can be subclassed outside this assembly.
```

Comment:
Please add a comment that internal types that can be instantiated from either this assembly or other assemblies which have InternalsVisibleTo access. For both these cases, there should be no security vulnerabilities introduced by overriding methods, hence these types can be ignored.